sdlib/d/gc/scanner.d

```diff
@@ -418,28 +418,35 @@ private:
 			}
 		}
 
-		if (pd.containsPointers) {
-			splitAndAddToWorklist(makeRange(e.address, e.address + e.size));
+		if (pd.containsPointers && e.usedCapacity >= PointerSize) {
+			splitAndAddToWorklist(
+				makeRange(e.address, e.address + e.usedCapacity));
 		}
 	}
 
 	void splitAndAddToWorklist(const(void*)[] range) {
 		assert(isAligned(range.ptr, PageSize),
 		       "Range is not aligned properly!");
-		assert(isAligned(range.length, PointerInPage),
-		       "Range has invalid size!");
 		assert(range.length > 0, "Cannot add empty range to the worklist!");
 
 		// Make sure we do not starve ourselves. If we do not have
 		// work in advance, then just keep some of it for ourselves.
 		if (cursor == 0) {
 			cursor = 1;
-			worklist[0] = range[0 .. PointerInPage];
-
-			range = range[PointerInPage .. range.length];
-			if (range.length == 0) {
+			if (range.length < PointerInPage * 2) {
+				worklist[0] = range;
 				return;
 			}
+
+			// Consume the tail portion of at least 1 page,
+			// possibly with a partial page on the end.
+			auto myWorkSize =
```

Comment:
myWorkSize sounds like one of these bad variable name from SEO optimized tutorial we find everywhere these days :)

Reply:
I found a way to calculate breakPoint in one call. Reads much nicer.